pkg/authenticator/authenticator.go

```diff
@@ -233,3 +243,38 @@ func resetIAMURL(auth Authenticator) bool {
 	}
 	return false
 }
+
+// retry ...
+func retry(logger *zap.Logger, authenticator Authenticator, retryfunc func() error) error {
+
+	retryGap := 4
+	var err error
+
+	for retryAttempt := 0; retryAttempt < maxRetryAttempt; retryAttempt++ {
+		err := retryfunc()
+		if err == nil {
+			return err
+		}
+
+		logger.Error("Error fetching fresh token", zap.Error(err), zap.Int("AttemptNo", retryAttempt+1))
+		// return if the error is not timeout
+		if !isTimeout(err) {
+			return err
+		}
+
+		time.Sleep(time.Second * time.Duration(retryGap))
+		retryGap = retryGap * 2
+		if retryGap > maxRetryGap {
+			retryGap = maxRetryGap
+		}
+	}
+
+	// Reset the IAM URL from to public, if it is private
```

Comment:
Typo
// Reset the IAM URL to public, if it is private
// Retry fetching IAM token again

Reply:
Updated